test/javascripts/turbolinks_test.coffee

```diff
@@ -185,6 +185,33 @@ describe 'Turbolinks', ->
           assertLinks(['foo.css'])
           done()
 
+    it 'many subsequent navigations do not break head asset tracking', (done) ->
```

Comment:
Unless we have a way to run Safari in CI, these don't seem valuable.  They probably give a false sense of confidence to the next people who come along and try to update the `head` query.